### PR TITLE
Fix/workaround to trigger reload of cc

### PR DESCRIPTION
**Pull request by tomwim**
### Description
When updating an old project the course controller might have missing scripts. A focus on the coursecontrollersetup is needed. This is a workaround to enforce this reload after scripts are loaded.

### CODE REVIEW

File: Runtime/CourseController/CourseControllerSetup.cs [hunk 1]
```diff
@@ -149,5 +150,23 @@ public void ResetToDefault()
...
+        [UnityEditor.Callbacks.DidReloadScripts]
+        private static void OnScriptsReloaded()
+        {
+            CourseControllerSetup setup = FindObjectOfType<CourseControllerSetup>();
+            List<Type> currentTypes = setup.GetComponents<Component>().Select(c => c.GetType()).ToList();
+            ICourseController cc = setup.GetCourseControllerFromType();
+            bool courseControllerHasMissingComponents = currentTypes.Except(cc.GetRequiredSetupComponents()).Any();
+            if (courseControllerHasMissingComponents)
+            {
+                Selection.activeObject = setup;
```
Comment (tomwim) [inline Runtime/CourseController/CourseControllerSetup.cs:167]: Calling Setup here is probably the better way. But did not think about it before 🤔

File: Runtime/CourseController/CourseControllerSetup.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 using System.Linq;
 using UnityEngine;
 using Innoactive.Creator.Core.Utils;
+using UnityEditor;
```
Comment (Gusinuhe) [inline Runtime/CourseController/CourseControllerSetup.cs:6]: It might break during build or execution.

**Review by Gusinuhe — CHANGES_REQUESTED** [review]
Still the same issue, an error message is thrown when playing the scene.

Test scenario:

An evaluator is trying to understand the creator, this person opens any example, plays it and finds the error and potentially no input.

File: Runtime/CourseController/CourseControllerSetup.cs (hunk 1 above)
Comment (tomwim) [reply Runtime/CourseController/CourseControllerSetup.cs:169]: Setup does not work

**Review by Gusinuhe — APPROVED** [review]

**Pull request merged**